### ISSUE TITLE
telemetryLoggerMiddleware does not log Adaptive Card responses

### ISSUE THREAD

**Issue by nathanlindorff** [microsoft/botbuilder-js#1574]
**Is your feature request related to a problem? Please describe.**
I am using the Virtual Assistant template which uses the telemetryLoggerMiddleware. After using it for a while, it became apparent when Adaptive cards are being used as a reply they are not being picked up by the telemetry logger. Only `text` and `speak` are being populated into the event properties in the [`fillSendEventProperties`](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder-core/src/telemetryLoggerMiddleware.ts#L229) method.

_[telemetryLoggerMiddleware.ts line 243](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder-core/src/telemetryLoggerMiddleware.ts#L243)_
```
if (activity.text && activity.text.trim()) {
    properties[this.telemetryConstants.textProperty] = activity.text;
}

if (activity.speak && activity.speak.trim()) {
    properties[this.telemetryConstants.speakProperty] = activity.speak;
}
```

**Describe the solution you'd like**
I would like TelemetryLoggerMiddleware, or a variant of it, to log attachments like Adaptive cards. This could either be by default if detected, or by configuration.

**Describe alternatives you've considered**
To work around this issue, I could make my own version of the telemetryLoggerMiddleware, and may need to do so in the short term. In my option it would be better to have it as a part of the framework, as using Adaptive Cards is a documented way of sending messages. These should be captured.

[enhancement]

**Comment by v-kydela** [comment]
@nathanlindorff - I'm a little confused because you have a link that's labeled `fillUpdateEventProperties` but it links to `fillReceiveEventProperties` and then you've posted a code snippet from `fillSendEventProperties`. Can you explain why?

**Comment by nathanlindorff** [comment]
PEBKAC cut and paste error sorry. I'll fix it up. It all should point to fillSendEventProperties

**Comment by nathanlindorff** [comment]
What I ended up doing in my scenario was subclassing TelemetryLoggerMiddleware and overriding `fillSendEventProperties`.

I added a new 'attachment' property to which I wrote the stringified version of the attachment. I also did some specific work to extract the text from the Adaptive card into a readable form. The latter was more specific to my cards. In retrospect unless there is a standard way to turn cards to text this may be the better option than framework changes. 

Perhaps a documentation update would be more appropriate, indicating that if Cards are used, the telemetry middleware doesn't log any details.

**Comment by v-kydela** [comment]
@nathanlindorff - It is curious that `fillSendEventProperties` provides an unused `telemetryProperties` parameter. I wonder if it's meant to be used by a subclass like yours, in which case you'd only need to override `onSendActivity`.

**Comment by v-kydela** [comment]
@daveta - I've assigned you to this issue so that you can shed some light on it and decide if you want to change anything. There doesn't seem to be anything for my team to do.

**Comment by nathanlindorff** [comment]
@v-kydela I was initially going to do something with `telemetryProperties` but the attachment was part of the Activity.

**Comment by daveta** [comment]
Adding @scheyal : not sure how attachments fits into the reporting strategy..

**Comment by garypretty** [comment]
Assigned myself - discussing with @scheyal

**Comment by garypretty** [comment]
@nathanlindorff I just wanted to provide you with an update.  We agree that tracking adaptive cards / attachments would be useful / required for bot telemetry. We have added this to our list of requirements for an all up telemetry review taking place right now.  We will update this ticket once we have this scoped and planned in for a release.  Right now I would suggest your workaround of implementing your own telemetry logger. Thanks for raising the issue.

**Comment by clearab** [comment]
@garypretty Can you please confirm if this is still on the list for R9?

**Comment by carlosscastro** [comment]
@garypretty can you please confirm? Thanks!

**Comment by garypretty** [comment]
@carlosscastro @clearab This is still TBC for R9. 

@scheyal consideration of adaptive cards and their inclusion in telemetry should be considered when we make the necessary changes to telemetry for adaptive (to ensure the schema is aligned for waterfall / adaptive).  Can you ensure this is tracked when the work is picked up by the adaptive team?  Happy to discuss.

**Comment by garypretty** [comment]
@nathanlindorff This has now been addressed in JS / C# for release 4.9, with tracking items for Python / Java referenced above. Thanks again for raising the feature request.

**Comment by nathanlindorff** [comment]
Thanks for the update Gary!
